elixir/apps/domain/lib/domain/policies/policy/query.ex

```diff
@@ -0,0 +1,24 @@
+defmodule Domain.Policies.Policy.Query do
+  use Domain, :query
+
+  def all do
+    from(policies in Domain.Policies.Policy, as: :policies)
+    |> where([policies: policies], is_nil(policies.deleted_at))
+  end
+
+  def by_id(queryable \\ all(), id) do
+    where(queryable, [policies: policies], policies.id == ^id)
+  end
+
+  def by_account_id(queryable \\ all(), account_id) do
+    where(queryable, [policies: policies], policies.account_id == ^account_id)
+  end
+
+  def by_actor_id(queryable \\ all(), actor_id) do
+    join(queryable, :inner, [policies: policies], ag in assoc(policies, :actor_group),
+      as: :actor_groups
+    )
+    |> join(:inner, [actor_groups: actor_groups], a in assoc(actor_groups, :actors), as: :actors)
```

Comment:
Just a small thing, it's common to move the first argument out of the pipe too, we think about it as a struct transformation steps:
```suggestion
    queryable
    |> join(:inner, [policies: policies], ag in assoc(policies, :actor_group),
      as: :actor_groups
    )
    |> join(:inner, [actor_groups: actor_groups], a in assoc(actor_groups, :actors), as: :actors)
```

Reply:
Good to know.  Thanks for the heads up!

Reply:
This has been updated